src/main/java/updater/DownloadFile.java

```diff
@@ -0,0 +1,129 @@
+package updater;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URLConnection;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * A class in charge of downloading file.
+ */
+public class DownloadFile {
+    private static final Logger logger = LogManager.getLogger(DownloadFile.class.getName());
+
+    public static final int CONNECTION_TIMEOUT = 15000;
+    public static final int READ_CONNECTION_TIMEOUT = 30000;
+
+    private final URI source;
+    private final File dest;
+    private long totalFileSizeInBytes;
+
+    private final Optional<Consumer<Double>> methodToCallOnDownload;
```

Comment:
This too. `onDownload`? Also, I don't think it needs to be optional. Callers can just pass a no-op `a -> {}`, or we can do that ourselves in a constructor overload.